storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,29 +1,535 @@
+            .url(client.url())
+            .version(client.version())
+            .build();
+    }
+
+    /**
+     * Creates a QueueAsyncClient that sends requests to the storage queue service at {@code endpoint}.
+     * Each service call goes through the {@code httpPipeline}.
+     *
+     * @param endpoint URL for the Storage Queue service
+     * @param httpPipeline HttpPipeline that the HTTP requests and response flow through
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(URL endpoint, HttpPipeline httpPipeline, String queueName) {
+        this.queueName = queueName;
+
+        this.client = new AzureQueueStorageBuilder().pipeline(httpPipeline)
+            .url(endpoint.toString())
+            .build();
+    }
+
+    /**
+     * Creates a builder that can configure options for the QueueAsyncClient before creating an instance of it.
+     *
+     * @return A new {@link QueueClientBuilder} used create QueueAsyncClient instances.
+     */
+    public static QueueClientBuilder builder() {
+        return new QueueClientBuilder();
+    }
+
+    /**
+     * @return the URL of the storage queue
+     */
+    public URL getQueueUrl() {
+        try {
+            return new URL(client.url());
+        } catch (MalformedURLException ex) {
+            throw new IllegalStateException("Queue URL is malformed");
+        }
+    }
+
+    /**
+     * Creates a new queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create a queue</p>
+     *
+     * @codesnippet com.azure.storage.queue.queueAsyncClient.create
+     *
+     * @return A response that only contains headers and response status code
+     * @throws StorageErrorException If a queue with the same name already exists in the queue service.
+     */
+    public Mono<VoidResponse> create() {
+        return create(null);
+    }
+
+    /**
+     * Creates a new queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create a queue with metadata "queue:metadataMap"</p>
+     *
+     * <pre>
+     * client.create(Collections.singletonMap("queue", "metadataMap"))
+     *     .subscribe(response -&gt; System.out.printf("Create completed with status code %d", response.statusCode()));
+     * </pre>
+     *
+     * @param metadata Metadata to associate with the queue
+     * @return A response that only contains headers and response status code
+     * @throws StorageErrorException If a queue with the same name and different metadata already exists in the queue service.
+     */
+    public Mono<VoidResponse> create(Map<String, String> metadata) {
+        return client.queues().createWithRestResponseAsync(queueName, null, metadata, null, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Permanently deletes the queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Delete a queue</p>
+     *
+     * @codesnippet com.azure.storage.queue.queueAsyncClient.delete
```

Comment:
None of these codesnippet blocks are valid - you are missing the opening and closing brace. Generate the JavaDoc and you'll see what I mean.